web/src/App.tsx

```diff
@@ -30,6 +38,15 @@ const SettingsModal = lazy(() => import('features/modals/SettingsModal'));
 const TimeControllerWrapper = lazy(() => import('features/time/TimeControllerWrapper'));
 
 const isProduction = import.meta.env.PROD;
+// Type for the URL parameters that determine app state
+export type RouteParameters = {
+  zoneId?: string;
+  urlTimeAverage?: TimeAverages;
+  urlDatetime?: string;
+};
+
+// Context for sharing route parameters with child components
+export const RouteContext = React.createContext<RouteParameters>({});
```

Comment:
Is creating RouteContext necessary? It looks like [useParams already uses context under the hood](https://github.com/remix-run/react-router/blob/5f34a96b44ba1bbec73045efd7bb57032d595687/packages/react-router/lib/hooks.tsx#L299)